torch/csrc/jit/runtime/interpreter.cpp

```diff
@@ -1249,10 +1262,16 @@ struct InterpreterStateImpl : c10::intrusive_ptr_target {
             // Move inputs to a separate stack
```

Comment:
instead of modifying the interpreter, you could reuse, e.g. https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/ParallelThreadPoolNative.cpp#L73 and
https://github.com/pytorch/pytorch/blob/master/torch/csrc/autograd/engine.cpp#L515 - some update (e.g. name) is needed for ThreadLocalDebugInfoBase, since it's not going to be only for debug info;
Btw, we've just coincidentally had a discussion re. this update, and, e.g. thought about the ability to pass some arbitrary data from higher-level python layers (e.g. logical name of distributed node) too